File: niddler-example-android/src/main/java/com/icapps/sampleapplication/api/ExampleJsonApi.java
package com.icapps.sampleapplication.api;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

/**
 * Created by maartenvangiel on 14/11/2016.
 */
public interface ExampleJsonApi {

    @GET("/posts")
    Call<List<Post>> getPosts();

    @GET("/posts/{id}")
    Call<Post> getPost(@Path("id") final int id);

}

File: niddler-example-android/src/main/java/com/icapps/sampleapplication/api/ExampleJsonApi.kt
package com.icapps.sampleapplication.api

import okhttp3.MultipartBody
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.*

/**
 * Created by maartenvangiel on 14/11/2016.
 */
interface ExampleJsonApi {

    @get:GET("/posts")
    val posts: Call<List<Post>>

    @GET("/posts/{id}")
    fun getPost(@Path("id") id: Int): Call<Post>

    @Multipart
    @POST("/posts")
    fun createPost(@Part message: MultipartBody.Part, @Part attachment: MultipartBody.Part): Call<ResponseBody>

}
